hal/test-headers/ticker_api_tests.h

```diff
@@ -0,0 +1,138 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2017-2017 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/** \addtogroup hal_specification */
+/** @{*/
+
+#ifndef MBED_US_TICKER_API_TESTS_H
+#define MBED_US_TICKER_API_TESTS_H
+
+#include "device.h"
+
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * \defgroup hal_specification_ticker Ticker
+ * Ticker specification and tests
+ *
+ * To run the microsecond ticker hal tests use the command:
+ *
+ *     mbed test -t <toolchain> -m <target> -n tests-mbed_hal-us_lp_ticker*
+ *
+ * # Defined behavior
+ * * The function ticker_init is safe to call repeatedly - Verified by test ::ticker_init_test
+ * * The function ticker_init resets the internal count and disables the ticker interrupt - Verified by test ::ticker_init_test
+ * * Ticker frequency is non-zero and counter is at least 8 bits - Verified by ::ticker_info_test
+ * * The ticker rolls over at (1 << bits) and continues counting starting from 0 - Verified by ::ticker_overflow_test
```

Comment:
The only option to test this is to wait until ticker rolls over. There is not function to modify  ticker count. When ticker is 32 bit long, then this scenario will execute over 70 minutes. Is this intended?

Reply:
I would skip this tests for devices which take more than a certain amount of time to overflow (~30 seconds or so?). This would allow 16 and 24 bit microsecond tickers and 16 bit lp tickers to be tested. It might be good to have the max test time as a define, so if need be the full 70 minute rollover could be tested locally (but not in CI) if need be.

Reply:
Thank you. This is a good solution, but still on some boards it might be hard to catch rollover moment to prove that after rollover ticker count is equal to 0.

Reply:
I agree, this isn't possible to do for some timers. For the low power ticker it should be straightforward to test this since you can read each tick. For the microsecond ticker this could be optional similar to the 32 bit long overflow scenario.